bundles/org.openhab.binding.ring/src/main/java/org/openhab/binding/ring/handler/AccountHandler.java

```diff
@@ -0,0 +1,634 @@
+import org.openhab.binding.ring.internal.RestClient;
+import org.openhab.binding.ring.internal.RingAccount;
+import org.openhab.binding.ring.internal.RingDeviceRegistry;
+import org.openhab.binding.ring.internal.RingVideoServlet;
+import org.openhab.binding.ring.internal.data.Profile;
+import org.openhab.binding.ring.internal.data.RingDevices;
+import org.openhab.binding.ring.internal.data.RingEvent;
+import org.openhab.binding.ring.internal.errors.AuthenticationException;
+import org.openhab.binding.ring.internal.errors.DuplicateIdException;
+import org.openhab.binding.ring.internal.utils.RingUtils;
+import org.openhab.core.OpenHAB;
+import org.openhab.core.config.core.Configuration;
+import org.openhab.core.library.types.DateTimeType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.net.NetworkAddressService;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.osgi.service.http.HttpService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonParseException;
+
+/**
+ * The {@link RingDoorbellHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Wim Vissers - Initial contribution
+ * @author Peter Mietlowski - oAuth upgrade and additional maintenance
+ * @author Ben Rosenblum - Updated for OH4 / New Maintainer
+ */
+
+@NonNullByDefault
+public class AccountHandler extends BaseBridgeHandler implements RingAccount {
+
+    private @Nullable ScheduledFuture<?> jobTokenRefresh = null;
+    private @Nullable ScheduledFuture<?> eventRefresh = null;
+    private @Nullable Runnable runnableVideo = null;
+    private @Nullable RingVideoServlet ringVideoServlet;
+    private @NonNullByDefault({}) HttpService httpService;
+    private final String thingId;
+
+    // Current status
+    protected OnOffType status = OnOffType.OFF;
+    protected OnOffType enabled = OnOffType.ON;
+    protected final Logger logger = LoggerFactory.getLogger(AccountHandler.class);
+
+    // Scheduler
+    protected @Nullable ScheduledFuture<?> refreshJob;
+
+    /**
+     * The user profile retrieved when authenticating.
+     */
+    private @NonNullByDefault({}) Profile userProfile;
+    /**
+     * The registry.
+     */
+    private @NonNullByDefault({}) RingDeviceRegistry registry;
+    /**
+     * The RestClient is used to connect to the Ring Account.
+     */
+    private @NonNullByDefault({}) RestClient restClient;
+    /**
+     * The list with events.
+     */
+    private @NonNullByDefault({}) List<RingEvent> lastEvents;
+    /**
+     * The index to the current event.
+     */
+    private int eventIndex;
```

Comment:
It looks that this is always `0`, it is set to `0` but never updated. Is that a flaw in the code`